src/dist/cache.rs

```diff
@@ -14,23 +17,33 @@ use util;
 
 use errors::*;
 
+#[derive(Clone, Debug)]
+pub struct CustomToolchain {
```

Comment:
It seems a little unfortunate to have two `CustomToolchain` structs that differ by only one member. I gather that since this one gets stored in a `HashMap` the other member is the key, but it's still not great.